core/trino-spi/src/main/java/io/trino/spi/block/AbstractRowBlock.java

```diff
@@ -65,6 +65,20 @@ protected AbstractRowBlock(int numFields)
         this.numFields = numFields;
     }
 
+    public void writeRowsTo(int[] positions, int positionsCount, BatchRowWriter batchWriter)
```

Comment:
Typically when dealing with positions array, the interfaces in SPI allow `int offset, int length` instead of just `int positionsCount` for flexibility.

Reply:
Ok, used offset, length